src/interpreter/inner.rs

```diff
@@ -95,11 +103,12 @@ pub enum Inner {
 // possible
 /// Parses an `Inner` and appropriate `Stack` from completed transaction data,
 /// as well as the script that should be used as a scriptCode in a sighash
-pub fn from_txdata<'txin>(
+/// Tr outputs don't have script code and return None.
+pub(super) fn from_txdata<'txin>(
     spk: &bitcoin::Script,
     script_sig: &'txin bitcoin::Script,
     witness: &'txin Witness,
-) -> Result<(Inner, Stack<'txin>, bitcoin::Script), Error> {
+) -> Result<(Inner, Stack<'txin>, Option<bitcoin::Script>), Error> {
```

Comment:
should the script_sig also be optional here then?

Reply:
That should be provided as `Script::new`. Every segwti tx has an empty script sig instead of the None script sig.